### ISSUE TITLE
Support Ruff formatter as an alternative for Black

### ISSUE THREAD

**Issue by jond01** [adamchainz/blacken-docs#352]
### Description

Ruff formatter can replace Black, and it's much faster:
https://docs.astral.sh/ruff/formatter/

I suggest supporting it as an option while keeping Black the default Python formatter.
Opting for Ruff can be done via a flag or persistent file configuration.

What do you think?

**Comment by adamchainz** [comment]
Yeah, I’d be tempted to add this and rename the project to `format-docs` or similar.

I think without a Ruff Python API, it wouldn't be particularly feasible. The only alternative would be to shell out to `ruff format` for each snippet, which would be slow due to process launch overhead (no matter how fast Ruff is).

I found the upstream Python API issue: https://github.com/astral-sh/ruff/issues/659

**Comment by gaborbernat** [comment]
Seem this can now be done via https://github.com/amyreese/ruff-api

**Comment by adamchainz** [comment]
Thanks for the link, but I’d rather not depend on a “highly experimental” third-party package.

**Comment by adamchainz** [comment]
Ruff plans to implement this functionality: https://github.com/astral-sh/ruff/issues/8237
